### PR TITLE
FLPATH-1249 - Update code for upstream notifications plugin

### PR DESCRIPTION
**Pull request by masayag**
This PR continues https://github.com/parodos-dev/serverless-workflows/pull/223.

In addition, it fixes https://issues.redhat.com/browse/FLPATH-1233

and solves https://issues.redhat.com/browse/FLPATH-1457

### CODE REVIEW

File: escalation/ticketEscalation.sw.yaml [hunk 1]
```diff
@@ -87,13 +87,14 @@ states:
...
           arguments:
-            title: '"ATTN: Escalation for ticket - " + .createResponse.ticketId'
-            message: '"Please manage escalation ticket " + .createResponse.ticketId '
-            origin: "Escalation Workflow"
-            topic: "Escalation Workflow"
-            actions:
-              - title: '"Escalation ticket " + .createResponse.ticketId'
-                url: .createResponse.browseUrl
+            recipients:
+              type: "broadcast"
```
Comment (ydayagi) [inline escalation/ticketEscalation.sw.yaml:91]: there's a story opened for this. https://issues.redhat.com/browse/FLPATH-1233. do u want to fix it here?

File: mta-v6.x/README.md [hunk 2]
```diff
@@ -10,14 +10,12 @@ the cloud-readiness compatibility of their code.
 
 # Inputs
 - `repositoryUrl` [mandatory] - the git repo url to examine
-- `backstageUser` [optional] - the backstage user to send backstage notification with the analysis results
-- `backstageGroup` [optional] - the backstage group to send backstage notification with the analysis results
+- `recipients` [mandatory] - A list of recipients for the notification in the format of `user:<namespace>/<username>` or `group:<namespace>/<groupname>`, i.e. `user:default/jsmith`.
```
Comment (gabriel-farache) [inline mta-v6.x/README.md:13]: `recipients` are not mandatory

File: mta-v7.x/README.md [hunk 3]
```diff
@@ -10,14 +10,12 @@ the cloud-readiness compatibility of their code.
 
 # Inputs
 - `repositoryUrl` [mandatory] - the git repo url to examine
-- `backstageUser` [optional] - the backstage user to send backstage notification with the analysis results
-- `backstageGroup` [optional] - the backstage group to send backstage notification with the analysis results
+- `recipients` [mandatory] - A list of recipients for the notification in the format of `user:<namespace>/<username>` or `group:<namespace>/<groupname>`, i.e. `user:default/jsmith`.
```
Comment (gabriel-farache) [inline mta-v7.x/README.md:13]: same

File: mta/README.md [hunk 4]
```diff
@@ -10,14 +10,12 @@ the cloud-readiness compatibility of their code.
 
 # Inputs
 - `repositoryUrl` [mandatory] - the git repo url to examine
-- `backstageUser` [optional] - the backstage user to send backstage notification with the analysis results
-- `backstageGroup` [optional] - the backstage group to send backstage notification with the analysis results
+- `recipients` [mandatory] - A list of recipients for the notification in the format of `user:<namespace>/<username>` or `group:<namespace>/<groupname>`, i.e. `user:default/jsmith`.
```
Comment (gabriel-farache) [inline mta/README.md:13]: same (I won't comment on all)

File: mta/mta.sw.yaml [hunk 5]
```diff
@@ -42,7 +42,7 @@ states:
         actionDataFilter:
           toStateData: ".applications"
     stateDataFilter:
-      output: "{backstageUser: .backstageUser, backstageGroup: .backstageGroup, repositoryURL: .repositoryURL, application: .repositoryURL as $r| if .applications == null then null else .applications|map(select(.repository.url == $r))[0] end}"
+      output: "{repositoryURL: .repositoryURL, application: .repositoryURL as $r| if .applications == null then null else .applications|map(select(.repository.url == $r))[0] end}"
```
Comment (gabriel-farache) [inline mta/mta.sw.yaml:45]: why not adding the `.recipients`? as you did https://github.com/parodos-dev/serverless-workflows/pull/297/files#diff-dc5c3c5cf797d197de6c902e605c4d693eb7b895ed40073983a8c66df0e30ac6R81

File: e2e/move2kube.sh [hunk 6]
```diff
@@ -42,8 +42,6 @@ GIT_ORG="gfarache31/m2k-test"
 GIT_REPO="bitbucket.org/${GIT_ORG}"
 GIT_SOURCE_BRANCH="master"
 GIT_TARGET_BRANCH="e2e-test-$(date +%s)"
-NOTIFICATIONS_DISABLED=true
```
Comment (gabriel-farache) [inline e2e/move2kube.sh:45]: I am fine with removing that but it is now less explicit that no notification will be sent and hence that we are not testing that part here (I know, it's already the case for MTA)

**Review by gabriel-farache — CHANGES_REQUESTED** [review]
Very nice, very big, some minor changes related to doc mainly

**Comment by gabriel-farache** [comment]
FYI, the move2kube E2E will never pass unless you create a PR with a branch on the upstream, this is for security reasons applied by @rgolangh (https://github.com/parodos-dev/serverless-workflows/commit/bfc6da83d53f61d798fe595361dff251f0f34df7) to avoid any person doing a fork and creating a PR to get access to the secrets, see https://stackoverflow.com/a/73867674

File: e2e/move2kube.sh (hunk 6 above)
Comment (masayag) [reply e2e/move2kube.sh:45]: is there a use case in which the notifications are enabled?

Comment (gabriel-farache) [reply e2e/move2kube.sh:45]: well, I think notifications are very important for users so from a ent-to-end standpoint,  testing that notifications are correctly sent and with good message and link would be great

That said, we are never testing them in e2e tests, I think that we are relying on QE to uncover issue with notifications

File: request-vm-cnv/application.properties [hunk 7]
```diff
@@ -1,6 +1,6 @@
 # Backstage Notifications service
 quarkus.rest-client.notifications.url=${BACKSTAGE_NOTIFICATIONS_URL:http://backstage-backstage.rhdh-operator}
-quarkus.openapi-generator.notifications.auth.bearertoken.bearer-token=${NOTIFICATIONS_BEARER_TOKEN}
+quarkus.openapi-generator.notifications.auth.BearerToken.bearer-token=${NOTIFICATIONS_BEARER_TOKEN}
```
Comment (gabriel-farache) [inline request-vm-cnv/application.properties:3]: not sure if case sensitive, did you test?
But better safe than sorry on this one :)

**Review by gabriel-farache — APPROVED** [review]

File: request-vm-cnv/application.properties (hunk 7 above)
Comment (masayag) [reply request-vm-cnv/application.properties:3]: it fails when there is a mismatch

Comment (gabriel-farache) [reply request-vm-cnv/application.properties:3]: thx

**Pull request closed without merging**